runtime/int.go

```diff
@@ -430,6 +441,20 @@ func intCheckedMul(v, w int) (int, bool) {
 	return x, true
 }
 
+func intCheckedPow(v, w int) (int, bool) {
+	// first try the shortcut - assuming the this operation is
```

Comment:
That's good enough for me. We can always tweak it later.

One other thing: doesn't it seem like on 64 bit systems there will be large integers that cannot be represented exactly by a float64 and therefore this won't work for some set of numbers?